Output file from sequence diagram

JackLeeHal opened the pull request:
Fixes #

Changes proposed in this pull request:
- Encode sequence diagram
- Send http request to plantuml
- Generate a png/svg/uml file

Commit:
Output file from sequence diagram
* Encode sequence diagram
* Send http request to plantuml
* Generate a png/svg/uml file

Commit:
Output file from sequence diagram
* Encode sequence diagram
* Send http request to plantuml
* Generate a png/svg/uml file

Commit:
Output file from sequence diagram
* Encode sequence diagram
* Send http request to plantuml
* Generate a png/svg/uml file

Commit:
Output file from sequence diagram
* Encode sequence diagram
* Send http request to plantuml
* Generate a png/svg/uml file

Commit:
Output file from sequence diagram
* Encode sequence diagram
* Send http request to plantuml
* Generate a png/svg/uml file

Commit:
Output file from sequence diagram
* Encode sequence diagram
* Send http request to plantuml
* Generate a png/svg/uml file

linwumeng commented on sysl2/sysl/diagutil.go:
```diff
@@ -0,0 +1,96 @@
+		logrus.Errorf("Unable to create file, Error:", err.Error())
+	}
+	defer f.Close()
+	jpeg.Encode(f, img, nil)
```
I understand  here png is converted to jpg, why?

JackLeeHal replied:
Fixed

linwumeng commented on sysl2/sysl/diagutil.go:
```diff
@@ -0,0 +1,96 @@
+	if b < 10 {
+		return byte(48 + b)
+	}
+	b -= 10
```
add empty line before

JackLeeHal replied:
Fixed

linwumeng commented on sysl2/sysl/diagutil.go:
```diff
@@ -0,0 +1,96 @@
+	if b < 26 {
+		return byte(65 + b)
+	}
+	b -= 26
```
add empty line before

JackLeeHal replied:
Fixed

linwumeng commented on sysl2/sysl/diagutil.go:
```diff
@@ -0,0 +1,96 @@
+	if b < 26 {
+		return byte(97 + b)
+	}
+	b -= 26
```
add empty line before

linwumeng commented on sysl2/sysl/diagutil.go:
```diff
@@ -0,0 +1,96 @@
+	if b == 1 {
+		return '_'
+	}
+	return '?'
```
add empty line before

JackLeeHal replied:
Fixed

linwumeng commented on sysl2/sysl/diagutil.go:
```diff
@@ -0,0 +1,96 @@
+}
+func encode6bit(b byte) byte {
+	if b < 10 {
+		return byte(48 + b)
```
add comment 48 -> '0'

JackLeeHal replied:
Fixed

Commit:
Sysl generate sequence diagram

* Parse parameters from command line
* Prepare struct data for sequenceDiagram

Commit:
Sysl generate sequence diagram

* Parse parameters from command line
* Prepare struct data for sequenceDiagram

Commit:
Sysl generate sequence diagram

* Parse parameters from command line

Commit:
Sysl generate sequence diagram

* Parse parameters from command line

linwumeng commented on sysl2/sysl/diagutil.go:
```diff
@@ -0,0 +1,96 @@
+	}
+	b -= 10
+	if b < 26 {
+		return byte(65 + b)
```
add comment 65 -> 'A'

JackLeeHal replied:
Fixed

linwumeng commented on sysl2/sysl/diagutil.go:
```diff
@@ -0,0 +1,96 @@
+	}
+	b -= 26
+	if b < 26 {
+		return byte(97 + b)
```
add comment 97 -> 'a'

JackLeeHal replied:
Fixed

linwumeng commented on sysl2/sysl/diagutil.go:
```diff
@@ -0,0 +1,96 @@
+	buf.WriteByte(encode6bit(c3 & 0x3F))
+	buf.WriteByte(encode6bit(c4 & 0x3F))
+}
+func encode6bit(b byte) byte {
```
add empty line before.
add comment to explain, like
6 bit makes value 0 to 63. The func maps 0-63 to characters '0'-'9', 'A'-'Z', 'a'-'z', '-', '_'. '?' should never be reached.

JackLeeHal replied:
Fixed

linwumeng commented on sysl2/sysl/diagutil.go:
```diff
@@ -0,0 +1,96 @@
+	return buf.String()
+}
+
+func encode3bytes(buf *bytes.Buffer, b1, b2, b3 byte) {
```
add comment to explain, like
3 bytes takes 24 bits. This splits 24 bits into 4 bytes of which lower 6-bit takes account.

JackLeeHal replied:
Fixed

Commit:
sequence diagram generator

* translate python code to golang
* add simple test

Commit:
sequence diagram generator

* translate python code to golang
* add simple test

Commit:
Output file from sequence diagram
* Fix comments
* Refactor code to be more like python version

Commit:
Output file from sequence diagram
* Fix comments
* Refactor code to be more like python version

Commit:
Output file from sequence diagram
* Fix comments
* Refactor code to be more like python version

Commit:
Output file from sequence diagram
* Fix comments
* Refactor code to be more like python version

Commit:
Output file from sequence diagram
* Fix comments
* Refactor code to be more like python version

Commit:
Output file from sequence diagram
* Fix comments
* Refactor code to be more like python version

Commit:
Output file from sequence diagram
* Fix comments
* Refactor code to be more like python version

Commit:
Output file from sequence diagram
* Add positive test cases

Commit:
Output file from sequence diagram
* Add positive test cases

Commit:
Output file from sequence diagram
* Add positive test cases

Commit:
Output file from sequence diagram
* Add positive test cases

Commit:
Output file from sequence diagram
* Add positive test cases

Commit:
Output file from sequence diagram
* Add positive test cases

Commit:
Output file from sequence diagram
* Add positive test cases

Commit:
Sysl generate sequence diagram

* Add format string parser func structure
* Finish the parser func

Commit:
Sysl generate sequence diagram

* Add format string parser func structure
* Finish the parser func

Commit:
Sysl generate sequence diagram

* Add format string parser func structure

Commit:
Sysl generate sequence diagram

* Add format string parser func structure
* Finish the parser func
* Merge with jeco's branch

Commit:
Sysl generate sequence diagram

* Add format string parser func structure
* Finish the parser func
* Merge with jeco's branch

Commit:
Sysl generate sequence diagram

* Add format string parser func structure
* Finish the parser func
* Merge with jeco's branch

Commit:
Sysl generate sequence diagram

* Add format string parser func structure
* Finish the parser func
* Merge with jeco's branch

Commit:
Output file from sequence diagram
* Refactor code
* Fix bug, should output .puml instead of .uml

Commit:
Output file from sequence diagram
* Refactor code
* Fix bug, should output .puml instead of .uml

Commit:
Output file from sequence diagram
* Refactor code
* Fix bug, should output .puml instead of .uml

Commit:
Output file from sequence diagram
* Refactor code
* Fix bug, should output .puml instead of .uml

Commit:
Output file from sequence diagram
* Refactor code
* Fix bug, should output .puml instead of .uml

Commit:
Refactor the code

* Apply visitor design pattern

Commit:
Sysl generate sequence diagram

* Add test case
* Fix code comments

Commit:
Sysl generate sequence diagram

* Add test case
* Fix code comments
* Decouple codes of sequence diagram

Commit:
Output file from sequence diagram
* Fix comments and refactor code

Commit:
Output file from sequence diagram
* Fix comments and refactor code

Commit:
Output file from sequence diagram
* Fix comments and refactor code

Commit:
Output file from sequence diagram
* Fix comments and refactor code

Commit:
Output file from sequence diagram
* Fix comments and refactor code

Commit:
Refactor the code

* Move to seqs package

Commit:
Refactor the code

* Move to seqs package
* Change seqs import

Commit:
Sequence diagram generate

* Add test case
* Add comments for variables unused

Commit:
Sequence diagram generate

* Add test case
* Add comments for variables unused

Commit:
Add test cases

Commit:
Output file from sequence diagram
* Fix test issue

Commit:
Output file from sequence diagram
* Fix test issue

Commit:
Output file from sequence diagram
* Fix test issue

Commit:
Output file from sequence diagram
* Fix test issue

Commit:
Sequence diagram generate

* Fix test case

Commit:
Sequence diagram generate

* Fix test case

Commit:
Sequence diagram generate

* Fix test case

Commit:
Merge remote-tracking branch 'loki/sd' into plantuml-output

Commit:
Sequence diagram generate

* Fix defects identified in unit test
* Add test cases
*

Commit:
Update Makefile to build sd binary

* Add next1-seqs command to build sd binary

Please DO export SYSL_PLANTUML=http://sysl-plantuml-default-dcx-tools.apps.omni.service.test
first, then run the command `make next1-seqs`

Commit:
Output file from sequence diagram
* Add negative test cases

Commit:
Merge remote-tracking branch 'loki/sd' into plantuml-output

# Conflicts:
#	sysl2/sysl/seqs.go
#	sysl2/sysl/seqs/diagutil_test.go
#	sysl2/sysl/seqs/utils_test.go
#	sysl2/sysl/seqs_test.go

Commit:
Output file from sequence diagram
* Remove unused file

Commit:
Output file from sequence diagram
* Fix the comments
